### PR TITLE
Remove `Partial<...>`

### PR DESCRIPTION
**Pull request by ronyeh**
Work in progress. :-)

I am testing out what it looks like if I remove usage of `Partial<...>`. I will employ `Required<...>` if needed.

### CODE REVIEW

File: src/chordsymbol.ts [hunk 1]
```diff
@@ -490,11 +490,11 @@ export class ChordSymbol extends Modifier {
   // ChordSymbol allows multiple blocks so we can mix glyphs and font text.
   // Each block can have its own vertical orientation
   // eslint-disable-next-line
-  getSymbolBlock(parameters?: any): any {
-    parameters = parameters ?? {};
```
Comment (rvilarl) [inline src/chordsymbol.ts:494]: @ronyeh I decided to go for this approach in order to handle null.

File: src/chordsymbol.ts [hunk 2]
```diff
@@ -547,8 +547,7 @@ export class ChordSymbol extends Modifier {
   // ### addText
   // Add a text block
   // eslint-disable-next-line
-  addText(text: string, parameters?: any): this {
-    parameters = parameters ?? {};
```
Comment (rvilarl) [inline src/chordsymbol.ts:551]: Same as above.

File: src/chordsymbol.ts [hunk 3]
```diff
@@ -579,8 +578,7 @@ export class ChordSymbol extends Modifier {
   }
 
   // eslint-disable-next-line
-  addGlyph(glyph: string, parameters?: any): this {
-    parameters = parameters ?? {};
```
Comment (rvilarl) [inline src/chordsymbol.ts:583]: Same

File: src/chordsymbol.ts [hunk 4]
```diff
@@ -592,8 +590,7 @@ export class ChordSymbol extends Modifier {
   // `addGlyphOrText("(+5#11)")`
   // will use text for the '5' and '11', and glyphs for everything else.
   // eslint-disable-next-line
-  addGlyphOrText(text: string, parameters?: any): this {
-    parameters = parameters ?? {};
```
Comment (rvilarl) [inline src/chordsymbol.ts:596]: Same

File: src/chordsymbol.ts (hunk 1 above)
Comment (ronyeh) [reply src/chordsymbol.ts:494]: I see. In these cases, is there evidence `null` is passed in as a parameter? I was just going for consistency with the rest of the code base, as I see other instances of `= {}` used as a default if the params is undefined.

File: src/chordsymbol.ts (hunk 2 above)
Comment (ronyeh) [reply src/chordsymbol.ts:551]: https://github.com/0xfe/vexflow/blob/master/src/system.ts

And it is used widely in Factory: https://github.com/0xfe/vexflow/blob/master/src/factory.ts

Is there something special about chordsymbol such that we should handle null as well? Or should we eventually adopt the `?? {}` in other parts of VexFlow? (I agree it is extra robust.)

**Comment by ronyeh** [comment]
Also, I'm still cleaning stuff up, so don't waste too much time reviewing the PR yet. :-)

I'll let you know when it's ready for your review.

**Comment by ronyeh** [comment]
Did your branch have visual diffs relative to the master?

I was able to remove all the Partials, and got to a decent stopping point for now.

**I see 2 visual diffs.**

One of them looks like a bug you fixed with `Stave.Vertical_Bar_Test`? It now draws a double vertical line.

The second diff is a little number 2 now shows up in the Volta tests: `Stave.Volta___Modifier_Measure_Test.Bravura.png` It looks like we fixed a bug. Is that something you fixed? Or is it something I accidentally fixed? :-D



I need to do a quick review tomorrow before giving you the go ahead for your review. This PR incorporates little fixes that I discovered along the way (while removing `Partial<...>`), so it covers a number of files. I can help walk you through some of them if necessary.

This PR also uncovered some inconsistencies with the way we handle fonts. We can address the font situation in a future PR.

We can decide how to proceed.

a) We can review + merge this into your branch, and then update your current PR to include these changes. 

b) We can ask Mohit to merge your current PR, and then I can submit this branch as a separate PR. (Or maybe more than one PR, if it is too tough to review as is.)

**Comment by rvilarl** [comment]
This branch is only fixing `Stave.Vertical_Bar_Test`.
The volta fix must have done by you. :) Please note that I fixed today the conflicts with main, pull them.

I would go for the b) approach. I believe that if I do not rebase, your commits will be acknowledged to you and it will be quicker to merge only one PR with Mohit. :)

Have a good day.

**Comment by ronyeh** [comment]
I don't mind who the code gets attributed to :-). My only goal is to create one or more clean PRs that Mohit is willing to merge. Once I'm done cleaning up this PR to your branch (and investigating the unexpected bug fix) I'll let you know and you can decide if you want to take any of the code and manually patch it into your PR. Like I said, I don't need any attribution. I just want to achieve consensus on our approach for options objects.

**Comment by ronyeh** [comment]
I think I'm at a good stopping point. 

I still need to investigate the unexpected bug fix.

```
Stave.Vertical_Bar_Test.Petaluma 372.165
Stave.Vertical_Bar_Test.Gonville 372.165
Stave.Vertical_Bar_Test.Bravura 372.165
Stave.Volta___Modifier_Measure_Test.Petaluma 0.0771369
Stave.Volta___Modifier_Measure_Test.Bravura 0.0587855
Stave.Volta___Modifier_Measure_Test.Gonville 0.0545349
```

We can decide which of my changes you'd like to manually merge into your branch. Then we can get Mohit to merge your PR before I submit the rest of the changes.

I might need to break up these changes into multiple PRs. :-) Sometimes I see things that need to be fixed and I can't help myself, haha.

**Comment by ronyeh** [comment]
FYI: I'm extracting some patch files from my branch and will send them to you. You can git apply the patches and we'll ask Mohit to merge. Then I'll rebase my branch on top of 0xfe's main branch.

**Comment by ronyeh** [comment]
I submitted my patch on top of your existing branch as a new PR: https://github.com/0xfe/vexflow/pull/1148/

https://github.com/0xfe/vexflow/pull/1148/commits

It includes all of your commits, so we can probably https://github.com/0xfe/vexflow/pull/1141 and also close this PR here.

I'll submit my extra changes (with Partial/Required and FontInfo handling) directly to 0xfe's repo.

**Pull request closed without merging**